tests/e2e/default/targets/target-aggregates.wdio-spec.js

```diff
@@ -458,39 +398,31 @@ describe('Target aggregates', () => {
       ];
 
-      await helper.takeScreenshot('detail-targets.png');
       await expectTargets(expectedTargets);
-      await openTargetDetails(expectedTargets[0].id);
+      await analytics.openTargetDetails(expectedTargets[0].id);
       await clickOnTargetAggregateListItem(clarissa._id);
-
-      expect(await element(by.css('.content-pane .meta h2')).getText()).toEqual('Clarissa');
+      expect(await contactsPageObject.getContactInfoName()).to.equal('Clarissa');
       // assert that the activity card exists and has the right fields.
-      expect(await element(by.css('.content-pane .meta > div > .card .action-header h3')).getText())
-        .toBe('Activity this month');
-      expect(await element.all(by.css('.content-pane .meta > div > .card .row label')).getText())
-        .toEqual(['Last updated', 'what a target!', 'the most target']);
-      expect(await element.all(by.css('.content-pane .meta > div > .card .row p')).getText())
-        .toEqual(['yesterday Clarissa', '40', '50%']);
-
-      await browser.navigate().back();
-      await helper.waitElementToPresentNative(element(by.css('.target-detail.card h2')));
-      await expectTargetDetails(expectedTargets[0]);
-
-      await openTargetDetails(expectedTargets[1].id);
+      expect(await contactsPageObject.getContactCardTitle()).to.equal('Activity this month');
+      validateCardField('Last updated', 'yesterday Clarissa');
+      validateCardField('what a target!', '40');
+      validateCardField('the most target', '50%');
+
+      await browser.back();
+      await analytics.expectTargetDetails(expectedTargets[0]);
+
+      await analytics.openTargetDetails(expectedTargets[1].id);
       await clickOnTargetAggregateListItem(prometheus._id);
 
-      expect(await element(by.css('.content-pane .meta h2')).getText()).toEqual('Prometheus');
+      expect(await contactsPageObject.getContactInfoName()).to.equal('Prometheus');
       // assert that the activity card exists and has the right fields.
-      expect(await element(by.css('.content-pane .meta > div > .card .action-header h3')).getText())
-        .toBe('Activity this month');
-      expect(await element.all(by.css('.content-pane .meta > div > .card .row label')).getText())
-        .toEqual(['Last updated', 'what a target!', 'the most target']);
-      expect(await element.all(by.css('.content-pane .meta > div > .card .row p')).getText())
-        .toEqual(['yesterday Prometheus', '18', '15%']);
-
-      await browser.navigate().back();
-      await helper.waitElementToPresentNative(element(by.css('.target-detail.card h2')));
-      await expectTargetDetails(expectedTargets[1]);
+      expect(await contactsPageObject.getContactCardTitle()).to.equal('Activity this month');
+      validateCardField('Last updated', 'yesterday Prometheus');
+      validateCardField('what a target!', '18');
+      validateCardField('the most target', '15%');
+
+      await browser.back();
```

Comment:
I think navigating to the specific place is better than doing browser back to avoid flaky tests.

Reply:
@latin-panda I understood from:
```
await browser.back();
await analyticsPage.expectTargetDetails(expectedTargets[0]);
 ```
that it validates if going back from contacts gets to the expected target details. Does this make sense?
I can get to the same place by replacing `await browser.back();` with:
```
await commonPage.goToAnalytics();
await analyticsPage.goToTargetAggregates(true);
await analyticsPage.openTargetDetails(expectedTargets[0].id);
```

Reply:
Ah okay, if it makes sense to test the navigation when going back, then it's fine. The second option you proposed is also okay. It depends on the purpose of the test.